engines/hypno/arcade.cpp

```diff
@@ -0,0 +1,323 @@
+				runIntro(video);
+			}
+			_nextLevel = arc.levelIfLose;
+			debugC(1, kHypnoDebugArcade, "Losing level and jumping to %s", _nextLevel.c_str());
+			break;
+		}
+
+		if (!arc.transitionVideo.empty() && background.decoder->getCurFrame() >= arc.transitionTime) {
+			debugC(1, kHypnoDebugArcade, "Playing transition %s", arc.transitionVideo.c_str());
+			arc.transitionTime = background.decoder->getFrameCount() + 1;
+			MVideo video(arc.transitionVideo, Common::Point(0, 0), false, false, false);
+			runIntro(video);
+			skipVideo(background);
+		}
+
+		if (!background.decoder || background.decoder->endOfVideo()) {
+			skipVideo(background);
+			if (!arc.winVideos.empty()) {
+				MVideo video(arc.winVideos.front(), Common::Point(0, 0), false, false, false);
+				runIntro(video);
+			}
+			_nextLevel = arc.levelIfWin;
+			debugC(1, kHypnoDebugArcade, "Wining level and jumping to %s", _nextLevel.c_str());
+			break;
+		}
+
+		if (shootSequence.size() > 0) {
+			ShootInfo si = shootSequence.front();
+			if (si.timestamp <= background.decoder->getCurFrame()) {
+				shootSequence.pop_front();
+				for (Shoots::iterator it = arc.shoots.begin(); it != arc.shoots.end(); ++it) {
+					if (it->name == si.name && it->animation != "NONE") {
+						Shoot s = *it;
+						s.video = new MVideo(it->animation, it->position, true, false, false);
+						playVideo(*s.video);
+						_shoots.push_back(s);
+						playSound(_soundPath + arc.enemySound, 1);
+					}
+				}
+			}
+		}
+
+		uint32 i = 0;
+		shootsToRemove.clear();
+
+		for (Shoots::iterator it = _shoots.begin(); it != _shoots.end(); ++it) {
+			if (it->video->decoder) {
+				int frame = it->video->decoder->getCurFrame();
+				if (frame > 0 && frame >= it->explosionFrame - 15 && !it->destroyed) {
+					hitPlayer();
+				}
+
+				if (frame > 0 && frame >= it->explosionFrame - 3 && !it->destroyed) {
+					skipVideo(*it->video);
+					//_health = _health - it->damage;
+				} else if (frame > 0 && frame >= it->video->decoder->getFrameCount()-2) {
+					skipVideo(*it->video);
+					shootsToRemove.push_back(i);
+				} else if (it->video->decoder->needsUpdate()) {
+					updateScreen(*it->video);
+				}
+			}
+			i++;
+		}
+		if (shootsToRemove.size() > 0) {
+			for (Common::List<uint32>::iterator it = shootsToRemove.begin(); it != shootsToRemove.end(); ++it) {
+				debugC(1, kHypnoDebugArcade, "Removing %d from %d size", *it, _shoots.size());
+				delete _shoots[*it].video;
+				_shoots.remove_at(*it);
+			}
+		}
+
+		if (_music.empty()) {
+			_music = _soundPath + arc.music;
+			playSound(_music, 1);
+		}
+
+		g_system->delayMillis(10);
+	}
+
+	// Deallocate shoots
+	for (Shoots::iterator it = _shoots.begin(); it != _shoots.end(); ++it) {
+		if (it->video->decoder)
+			skipVideo(*it->video);
+		delete it->video;
+	}
+
+	if (background.decoder)
+		skipVideo(background);
+
+}
+
+int HypnoEngine::detectTarget(Common::Point mousePos) {
+	int i = -1;
```

Comment:
again minor style (here and in other methods): reducing the scope of the variables increases the readability imo (and would allow to make them `const`)

Reply:
Which variable exactly? `i` ?

Reply:
I think this is done, please let me know if the scope of `i` can be reduced even more.